File: src/main/webapp/resources/js/services/EmpresaService.js
angular.module('PedeAgua').factory('empresaService', ['$http', '$q', function($http, $q) {

    var _salvarEmpresaUsuario = function(empresaUsuario) {
        return $http.post('http://localhost:8080/spring4-1/info/addEmpresaUsuario', empresaUsuario);
    };

    var _getEmpresas = function() {
        return $http.get('http://localhost:8080/spring4-1/info/getAllEmpresas');
    };

    var _getEmpresaPorCodigo = function(cod_empresa) {
        return $http.get('http://localhost:8080/spring4-1/info/getEmpresaPorCodigo?cod_empresa='+cod_empresa);
    };

    var _getEmpresaPorEmail = function(email) {
        return $http.get('http://localhost:8080/spring4-1/info/getEmpresaPorEmail?email='+email);
    };

    var _empresaLogin = function(email, senha) {
        return $http.get('http://localhost:8080/spring4-1/info/empresaLogin?email='+email+'&senha='+senha);
    };

    var _deslogarEmpresa = function(chave, valor) {
        return $http.get('http://localhost:8080/spring4-1/info/deslogarempresa?chave='+chave+'&valor='+valor);
    };

    return {
        getEmpresas: _getEmpresas,
        getEmpresaPorCodigo: _getEmpresaPorCodigo,
        salvarEmpresaUsuario: _salvarEmpresaUsuario,
        empresaLogin: _empresaLogin,
        deslogarEmpresa: _deslogarEmpresa,
        getEmpresaPorEmail: _getEmpresaPorEmail
    }


}]);

File: src/main/java/com/pedeagua/service/EmpresaService.java
package com.pedeagua.service;

import java.util.List;

import org.json.JSONArray;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.pedeagua.dao.IEmpresaDAO;
import com.pedeagua.entity.Empresa;
import com.pedeagua.entity.EmpresaUsuario;
import com.pedeagua.entity.Vendedor;
@Service
public class EmpresaService implements IEmpresaService {
	@Autowired
	private IEmpresaDAO empresaDAO;
	@Override
	public Empresa getEmpresaById(int cod_empresa) {
		Empresa obj = empresaDAO.getEmpresaById(cod_empresa);
		return obj;
	}	
	
	@Override
	public List<Empresa> getAllEmpresas(){
		return empresaDAO.getAllEmpresas();
	}
	@Override
	public synchronized boolean addEmpresa(Empresa empresa){
       if (empresaDAO.empresaExists(empresa.getEmail())) {
    	   return false;
       } else {
    	   empresaDAO.addEmpresa(empresa);
    	   return true;
       }
	}
	@Override
	public void updateEmpresa(Empresa empresa) {
		empresaDAO.updateEmpresa(empresa);
	}
	@Override
	public void deleteEmpresa(int cod_empresa) {
		empresaDAO.deleteEmpresa(cod_empresa);
	}
	@Override
	public boolean empresaUsuarioExists(String email, String senha) {
		// TODO Auto-generated method stub
		return empresaDAO.empresaUsuarioExists(email, senha);
	}
	@Override
	public Empresa getEmpresaByEmail(String email) {
		// TODO Auto-generated method stub
		return empresaDAO.getEmpresaByEmail(email);
	}
	@Override
	public boolean addEmpresaUsuario(EmpresaUsuario empresaUsuario) {
		
		if (empresaDAO.empresaExists(empresaUsuario.getEmail())) {
			boolean status = empresaDAO.updateEmpresaUsuario(empresaUsuario);//updateProdutoPorEmpresa(produto);		
			return status;
	       } else {
	    	   empresaDAO.addEmpresaUsuario(empresaUsuario);
	    	   return true;
	       }
	}
	@Override
	public boolean updateEmpresaUsuario(EmpresaUsuario empresaUsuario) {		
		return empresaDAO.updateEmpresaUsuario(empresaUsuario);
	}

	@Override
	public List<Vendedor> getEmpresasPorCoordenadas(String latitude, String longitude, JSONArray codigos) {
		
		return empresaDAO.getEmpresasPorCoordenadas(latitude, longitude, codigos);
	}
}
